hardhat.config.ts

```diff
@@ -0,0 +1,158 @@
+import { utils } from "ethers";
+import Web3 from "web3";
+
+dotenvConfig({ path: resolve(__dirname, "./.env") });
+
+const chainIds = {
+  ganache: 1337,
+  goerli: 5,
+  hardhat: 31337,
+  kovan: 42,
+  mainnet: 1,
+  rinkeby: 4,
+  ropsten: 3,
+  avalanche: 43114,
+  polygon: 137,
+  arbitrum: 42161,
+};
+
+const alchemyApiKey = process.env.ALCHEMY_API_KEY;
+if (!alchemyApiKey) {
+  throw new Error("Please set your ALCHEMY_API_KEY in a .env file");
+}
+
+let PRIVATE_KEY;
+const ETHERSCAN_API = process.env.ETHERSCAN_API_KEY;
+const POLYGONSCAN_API = process.env.POLYGON_API_KEY;
+const ARBISCAN_API = process.env.ARBISCAN_API_KEY;
+const SNOWTRACE_API = process.env.SNOWTRACE_API_KEY;
+const mnemonic = process.env.MNEMONIC;
+
+if(!mnemonic) {
+  PRIVATE_KEY = process.env.PRIVATE_KEY
+}
+
+function createTestnetConfig(
+  network: keyof typeof chainIds
+): NetworkUserConfig {
+  const url: string =
+    "https://eth-" + network + ".alchemyapi.io/v2/" + alchemyApiKey;
+  return {
+    accounts: {
+      count: 10,
+      initialIndex: 0,
+      mnemonic,
+      path: "m/44'/60'/0'/0",
+    },
+    chainId: chainIds[network],
+    url,
+  };
+}
+
+function getNetworkUrl(networkType: string) {
+  //console.log(process.env);
+  if (networkType === "avalanche")
+    return "https://api.avax.network/ext/bc/C/rpc";
+  else if (networkType === "polygon")
+    return `https://polygon-mainnet.g.alchemy.com/v2/${alchemyApiKey}`;
+  else if (networkType === "arbitrum")
+    return `https://arb-mainnet.g.alchemy.com/v2/${alchemyApiKey}`;
+  else return `https://eth-mainnet.alchemyapi.io/v2/${alchemyApiKey}`;
+}
+
+function getScanApiKey(networkType: string) {
+  if (networkType === "avalanche") return SNOWTRACE_API;
+  else if (networkType === "polygon") return POLYGONSCAN_API;
+  else if (networkType === "arbitrum") return ARBISCAN_API;
+  else return ETHERSCAN_API;
+}
+
+/**
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+const config: HardhatUserConfig = {
+  solidity: {
+    compilers: [
+      {
+        version: "0.7.6",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+      {
+        version: "0.6.0",
+      },
+      {
+        version: "0.6.2",
+      },
+      {
+        version: "0.6.5",
+      },
+    ],
+  },
+  networks: {
+    hardhat: {
+      accounts: {
+        mnemonic,
+      },
+      chainId: chainIds.hardhat,
+      forking: {
+        url: String(getNetworkUrl(String(process.env.networkType))),
+      },
+    },
+    goerli: createTestnetConfig("goerli"),
+    kovan: createTestnetConfig("kovan"),
+    rinkeby: createTestnetConfig("rinkeby"),
+    ropsten: createTestnetConfig("ropsten"),
+    mainnet: {
+      url: getNetworkUrl("mainnet"),
+    },
+    polygon: {
+      url: getNetworkUrl("polygon"),
+    },
+    avalanche: {
+      url: getNetworkUrl("avalanche"),
+    },
+    arbitrum: {
+      url: getNetworkUrl("arbiturm"),
+    },
```

Comment:
@pradyuman-verma I think we are not passing accounts or private key? For these networks? Because we are gonna use these network for deployed

Reply:
Okay we can, But then we have to provide a private key as a required parameter, otherwise the config will prompt errors

Reply:
I think we can control that if we have `process.env.PRIVATE_KEY` then only we are gonna pass it?

Reply:
```js
accounts: [`0x${PRIVATE_KEY}`],
```
then if PRIVATE_KEY = undefined.
I think then there will be error

Reply:
![image](https://user-images.githubusercontent.com/63545809/145648706-5696ebc9-032a-4e14-ba76-2a0f3308ee6a.png)

Reply:
![image](https://user-images.githubusercontent.com/63545809/145648759-c489bf11-45cf-4b83-84b0-9769c5f3dd53.png)

Reply:
I think what we can do is provide a dummy PRIVATE_KEY value.
```js
const PRIVATE_KEY = 0x{32 byte string};
if(process.env.PRIVATE_KEY){
PRIVATE_KEY = process.env.PRIVATE_KEY
}
```

Reply:
Sure! I think we can do like that:

```
let PRIVATE_KEY = 0x00....000000dead;
if(process.env.PRIVATE_KEY){
   PRIVATE_KEY = process.env.PRIVATE_KEY
}
```

Reply:
Or we can do like this:

```
...
accounts: !!PRIVATE_KEY ? [`0x${PRIVATE_KEY}`] : [],
```

Reply:
Okay, we are Implementing it.

Reply:
```
accounts: !!PRIVATE_KEY ? [`0x${PRIVATE_KEY}`] : mnemonic,
```
like this right !!